src/plugin/config_test.go

```diff
@@ -33,3 +32,20 @@ func TestFetchConfigFromEnvironment(t *testing.T) {
 	require.NoError(t, err, "fetch should not error")
 	assert.Equal(t, "test-message", config.Message, "fetched message should match environment")
 }
+
+func unsetEnv(t *testing.T, key string) {
```

Comment:
@jamestelfer do we not want to run this via a `defer` in cases where we set an environment variable? I thought we wanted to ensure global state was cleaned up after running each test.

Reply:
Regardless, I can port this change into `ecs-task-runner-buildkite-plugin` once this is merged.

Reply:
Good question. The [`t.Cleanup`](https://pkg.go.dev/testing#T.Cleanup) method gets called after a test is executed. It's similar to defer, but operates at the test scope rather than the function scope, making it easier to use in helper functions.

This is how `t.Setenv()` works under the hood.

Reply:
Also: I realise that I gave you a bad steer in my initial comment in suggesting a code search.

I didn't think a good example would be so hard to find, but when I did the search I suggested it was a lot harder to find a good example than I thought :( And the best example I found (in Silk) wasn't as good as it could be.

Reply:
(This is why I contributed a change rather than commenting: I felt like I'd led you up the garden path, sorry)

Reply:
All good, I will incorporate this "upstream".